tests/foreman/cli/test_logging.py

```diff
@@ -0,0 +1,189 @@
+
+:CaseLevel: Acceptance
+
+:CaseComponent: logging
+
+:TestType: Functional
+
+:CaseImportance: Medium
+
+:Upstream: No
+"""
+import re
+import logging
+from robottelo import ssh
+from robottelo.test import CLITestCase
+from robottelo.cli.factory import (
+    make_org,
+    make_product_wait,
+    make_repository,
+)
+from fauxfactory import gen_string
+log = logging.getLogger(__name__)
+
+
+class SimpleLoggingTestCase(CLITestCase):
+    """Test class for default logging to files."""
+
+    org = None
+    product = None
+
+    def setUp(self):
+        """Tests for logging to files"""
+        super(SimpleLoggingTestCase, self).setUp()
+
+        # need own org for the manifest refresh test
+
+        if SimpleLoggingTestCase.org is None:
+            SimpleLoggingTestCase.org = make_org(cached=True)
+        if SimpleLoggingTestCase.product is None:
+            SimpleLoggingTestCase.product = make_product_wait(
+                {u'organization-id': SimpleLoggingTestCase.org['id']},
+            )
+
+    def _make_repository(self, options=None):
+        """Makes a new repository and asserts its success"""
+        if options is None:
+            options = {}
+
+        if options.get('product-id') is None:
+            options[u'product-id'] = self.product['id']
+
+        return make_repository(options)
+
+    def test_positive_logging_from_foreman_core(self):
+        """Check that GET command to Hosts API is logged and has request ID."""
+        GET_line_found = 0
+        # log the start of the test in case of problems with slow systems and time zones
+        self.logger.info("Testing test_positive_logging_from_foreman_core")
+        with ssh.get_connection() as connection:
+            result = connection.run('hammer host list')
+            self.assertEqual(result.return_code, 0)
+            # extract last ten lines from log
+            result = connection.run('tail -20 /var/log/foreman/production.log > \
+                        /var/tmp/logfile_from_foreman_core')
+            self.assertEqual(result.return_code, 0)
+        # use same location on remote and local for log file extract
+        logfile_location = '/var/tmp/logfile_from_foreman_core'
+        ssh.download_file(logfile_location)
+        # search the log file extract for the line with GET to host API
+        with open("/var/tmp/logfile_from_foreman_core", "r") as logfile:
+            for line in logfile:
+                if re.search(r'Started GET \"\/api/hosts\?page=1', line):
+                    self.logger.info('Found the line with GET to hosts API')
+                    GET_line_found = 1
+                    # Confirm the request ID was logged in the line with GET
+                    match = re.search(r'\[I\|app\|\w{8}\]', line)
+                    if match:
+                        self.logger.info("Request ID found for logging from foreman core")
+                    else:
+                        assert match == 1, "Request ID not found"
```

Comment:
I find this code hard to understand. Since `re.search()` will return None if match was not found, in else branch `match` will be None, meaning this assertion can never succeed. I guess the goal is to fail test with specific message? Then it's better to use [`self.fail()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.fail).

If you want to fail test if `re.search()` doesn't find anything, then perhaps it's better to refactor it a little:

```python
match = re.search(r'\[I\|app\|\w{8}\]', line)
assert match, "Request ID not found"
# `match` is guaranteed to be proper Match object
self.logger.info("Request ID found for logging from foreman core")
```

Also, you probably want to call `return` or `break` after finding the match, as there is no point in analyzing subsequent lines.

Reply:
Thank you, I have now fixed two sections like that as you suggested,